primitives/runtime/src/traits.rs

```diff
@@ -625,13 +625,32 @@ pub trait Dispatchable {
 	type Origin;
 	/// ...
 	type Trait;
+	/// An opaque set of information attached to the transaction. This could be constructed anywhere
+	/// down the line in a runtime. The current Substrate runtime uses a struct with the same name
+	/// to represent the dispatch class and weight.
+	type Info;
 	/// Additional information that is returned by `dispatch`. Can be used to supply the caller
 	/// with information about a `Dispatchable` that is ownly known post dispatch.
 	type PostInfo: Eq + PartialEq + Clone + Copy + Encode + Decode + Printable;
 	/// Actually dispatch this call and return the result of it.
 	fn dispatch(self, origin: Self::Origin) -> crate::DispatchResultWithInfo<Self::PostInfo>;
 }
 
+/// Shortcut to reference the `Info` type of a `Dispatchable`.
+pub type DispatchInfoOf<T> = <T as Dispatchable>::Info;
+/// Shortcut to reference the `PostInfo` type of a `Dispatchable`.
+pub type PostDispatchInfoOf<T> = <T as Dispatchable>::PostInfo;
+
+impl Dispatchable for () {
```

Comment:
Can we use this somewhere? All the types are just `()`?

Why not take the type you created for balances and system and make it generic? Taking, the Origin, Trait, Info and PostInfo as generic arguments.

We can even go that far that we provide `()` as default values for these generic parameters.

Reply:
But we want to keep the plain impl on `()`, right? Because when _replacing_ it by a generic type we still need to touch *every* test to use that new type.

Reply:
Yeah sounds reasonable. i did not thought about this.

Reply:
I think we cannot really improve here without requiring all tests to use something like a (generic) `TestCall` instead of of `()`. If we are are willing to do that we were able to unify this to a generic type and also add the full bound to `frame_system::trait::Call` that is `Dispatchable<Info=DispatchInfo, PostInfo=PostDispatchInfo>` because we do not need to put a impl for `()` in primitives. 

I am not sure if it is worth the effort right now given that we want to move `Dispatchable` in the future.

Do you think I should still add this generic type to primitives and use it from the balances tests that currently use their own `Call` struct for testing? I feel it is not worth the effort when it is only one module that needs it.

Reply:
The generic type is already used in 2 places.^^ If we don't need it in Polkadot for now, fine, don't move it. :)

Reply:
Yes I would like to leave it for now. Polkadot does not need it. Just submitted the companion. Thanks for the feedback.